crates/typst/src/image/raster.rs

```diff
@@ -137,3 +155,65 @@ fn format_image_error(error: image::ImageError) -> EcoString {
         err => eco_format!("failed to decode image ({err})"),
     }
 }
+
+/// Encodes an image using a thread pool.
+fn encode_image(
+    dynamic: &Arc<image::DynamicImage>,
+    is_pdf: bool,
+    format: RasterFormat,
+) -> Option<Arc<OnceCell<Bytes>>> {
+    // If we're not rendering to a PDF we never run.
+    // This also means we're never instantiating the thread pool.
+    if !is_pdf {
+        return None;
+    }
+
+    if format == RasterFormat::Jpg && matches!(&**dynamic, DynamicImage::ImageLuma8(_)) {
+        return None;
+    }
+
+    if format == RasterFormat::Jpg && matches!(&**dynamic, DynamicImage::ImageRgb8(_)) {
+        return None;
+    }
+
+    // Special case for encoding luma images.
+    let out = Arc::new(OnceCell::new());
+    if format == RasterFormat::Png && matches!(&**dynamic, DynamicImage::ImageLuma8(_)) {
```

Comment:
what is &**? is it possible to use take the inner data out of ImageLuma8 and avoid the unreachable!() in below?